motor/core.py

```diff
@@ -140,9 +140,18 @@ def __init__(self, *args, **kwargs):
         self._io_loop = io_loop
 
         kwargs.setdefault("connect", False)
-        kwargs.setdefault(
-            "driver", DriverInfo("Motor", motor_version, self._framework.platform_info())
-        )
+
+        driver_info = DriverInfo("Motor", motor_version, self._framework.platform_info())
+
+        if kwargs.get("driver"):
+            provided_info = kwargs.get("driver")
```

Comment:
We should check the type here to give a nice error message in this case `Client(driver="string")`